llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -3302,14 +3302,16 @@ static bool isAllocSiteRemovable(Instruction *AI,
 
         if (getFreedOperand(cast<CallBase>(I), &TLI) == PI &&
```

Comment:
I'd prefer to add a `Family &&` here. No need to do the other checks if we don't have an allocation family in the first place.